### PR TITLE
Bumped dependency versions

### PR DESCRIPTION
**Pull request by pranav-bhatt**
The main motivation behind this was that the`tokio` version being used in this crate was still `0.2` and this was causing major issues in my project. Refactored it to use `tokio 1.7*`, and also ran `cargo fmt` (Do tell me if this might cause issues).

I also wanted to ask if this warrants a version bump since the change in tokio versions might cause issues for others.

### CODE REVIEW

**Comment by pranav-bhatt** [comment]
Some of the tests failed:
```
failures:
[00:02:58] 
[00:02:58] ---- observer::test::test_observe stdout ----
[00:02:58] thread 'main' panicked at 'assertion failed: `(left == right)`
[00:02:58]   left: `[79, 75]`,
[00:02:58]  right: `[100, 97, 116, 97, 49]`', src\observer.rs:480:26
[00:02:58] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[00:02:58] 
[00:02:58] ---- observer::test::test_unobserve stdout ----
[00:02:58] thread 'main' panicked at 'assertion failed: `(left == right)`
[00:02:58]   left: `[79, 75]`,
[00:02:58]  right: `[100, 97, 116, 97, 49]`', src\observer.rs:525:17
[00:02:58] 
[00:02:58] ---- server::test::multicast_server_all_coap stdout ----
[00:02:58] thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 10060, kind: TimedOut, message: "A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." }', src\server.rs:613:44
[00:02:58] 
[00:02:58] ---- server::test::test_echo_server_no_token stdout ----
[00:02:58] thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 10060, kind: TimedOut, message: "A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." }', src\server.rs:494:44
[00:02:58] 
[00:02:58] ---- server::test::test_update_resource stdout ----
[00:02:58] thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 10060, kind: TimedOut, message: "A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." }', src\server.rs:540:26
```

**Comment by Covertness** [comment]
It seems the payload was changed

File: src/server.rs [hunk 1]
```diff
@@ -297,7 +320,7 @@ impl Stream for CoAPServer {
     type Item = Result<Message, io::Error>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        if let Poll::Ready(Some((p, a))) = self.receiver.poll_next_unpin(cx) {
+        if let Poll::Ready(Some((p, a))) = self.receiver.poll_recv(cx) {
```
Comment (pranav-bhatt) [inline src/server.rs:323]: So I narrowed down the problematic code to this line. This was a silly mistake done by me, but I'm not able to understand how to move forward. Essentially the `CoAPServer.receiver` is currently of the type `UnboundedReceiver`, but needs to be wrapped by `UnboundedReceiverStream`. I tried changing the types, but I was running into bizarre issues. 

@Covertness I'm sure you're quite busy, but is it possible for you to take a look at this? Hopefully it won't take much time

**Comment by pranav-bhatt** [comment]
> It seems the payload was changed

So it seems the response that is being received is the one from the `response_handler`, i.e, b"OK". I've spent the entire day trying to fix this, but I'm not even sure how in the original repo this test case was passing. Shouldn't the `client.receive()` be getting "OK" as the response always?

Also I'm basically out of ideas, but I feel this is a silly fix. Would you be able to take a look? Hopefully shouldn't take too long :)

**Comment by Covertness** [comment]
The problem happened on [this line](https://github.com/pranav-bhatt/coap-rs/blob/master/src/server.rs#L328). The socket always produce `Poll::Ready` result after the first frame reached. It maybe caused by the new version of `tokio_util::udp::UdpFramed`

**Comment by pranav-bhatt** [comment]
Are you a part of a IRC or Slack server so I could ping you there?
> 
> 
> The problem happened on [this line](https://github.com/pranav-bhatt/coap-rs/blob/master/src/server.rs#L328). The socket always produce `Poll::Ready` result after the first frame reached. It maybe caused by the new version of `tokio_util::udp::UdpFramed`

I'll try downgrading this and check.

**Comment by pranav-bhatt** [comment]
> I'll try downgrading this and check.

Downgrading to a version below 0.6 unfortunately breaks stuff. Guess we'll have to figure out another way.

**Comment by Covertness** [comment]
The new version has [its own Stream](https://tokio.rs/tokio/tutorial/streams). I think it will be compatible with the socket of new version.

**Comment by pranav-bhatt** [comment]
> The new version has [its own Stream](https://tokio.rs/tokio/tutorial/streams). I think it will be compatible with the socket of new version.

Do correct me if I'm wrong, but from what I can see, the `Stream` trait is just a reexport of the [`futures::stream::Stream`](https://docs.rs/futures-core/0.3.15/futures_core/stream/trait.Stream.html) trait, so I don't think this is the issue. Quoting [`tokio_stream::StreamExt`](https://docs.rs/tokio-stream/0.1.6/tokio_stream/trait.StreamExt.html):

> Be aware that the Stream trait in Tokio is a re-export of the trait found in the futures crate, however both Tokio and futures provide separate StreamExt utility traits, and some utilities are only available on one of these traits. Click here to see the other StreamExt trait in the futures crate. 
>
> If you need utilities from both StreamExt traits, you should prefer to import one of them, and use the other through the fully qualified call syntax.

**Comment by Covertness** [comment]
Okay. Waiting for the reply from the tokio team.

**Comment by pranav-bhatt** [comment]
> Okay. Waiting for the reply from the tokio team.

@Covertness  I had a couple of other issues, and I was hoping that we could communicate real-time over a chat service or something else. Could I possibly have your slack or twitter handle?

**Comment by pranav-bhatt** [comment]
Can we have a new release? *v1.0 wink wink*

**Comment by Covertness** [comment]
Sorry for not getting back to you sooner. So busy recently. You can touch me by email if you want communicate with me. My email is wuyingfengsui@gmail.com.
Thank you for your pr. I will release it.

**Pull request merged**